### ISSUE TITLE
[Feature request]: Report distro name in useragent

### ISSUE THREAD

**Issue by razzeee** [flatpak/flatpak#5549]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

The user agent already reports flatpak version and ostree version, additionally, reporting the distro would be nice.

I would expect that in the form of something like this:
```
libostree/2022.2 flatpak/1.12.7 distro/Fedora
```
or

```
libostree/2022.2 flatpak/1.12.7 distro/Fedora 39
```

**Comment by nanonyme** [comment]
Should this be using os-release format only or arbitrary per-distro specifications as data source?

**Comment by razzeee** [comment]
I'm not sure if I can answer that correctly

**Comment by smcv** [comment]
> reporting the distro would be nice

Nice for whom, and for what purpose?

**Comment by razzeee** [comment]
I think mostly where we should spend more time on documentation/moving people over to newer flatpak versions etc

So just as a way to know which distros might need help or a reminder to ship newer versions.

**Comment by flexagoon** [comment]
I don't think this has much value, since Flatpak is supposed to be a distro-agnosic platform to run apps on.

> So just as a way to know which distros might need help or a reminder to ship newer versions.

Do you mean the version of flatpak apps or of the flatpak package itself? If you mean apps, their versions are not related to distros in any way. If you mean flatpak itself, you can already see distros that ship an old version [here](https://repology.org/project/flatpak/versions).

**Comment by nanonyme** [comment]
> Do you mean the version of flatpak apps or of the flatpak package itself? If you mean apps, their versions are not related to distros in any way. If you mean flatpak itself, you can already see distros that ship an old version [here](https://repology.org/project/flatpak/versions).
It would definitely provide some value to see actual usage data pinned to said distros and versions. Currently repo maintainers can mostly see which versions of flatpak are still widely used.

**Comment by freechelmi** [comment]
That could be very useful indeed to measure how flatpak progress on blocking distro like Ubuntu

**Comment by cassidyjames** [comment]
I was approached by a large ISV (one of the top 10 most-downloaded apps from Flathub) asking if we (Flathub) could provide the number of installs/updates on Steam Deck, as they are evaluating increasing their engineering budget around Steam Deck and thus Linux. While I can point them to the stats for total Flathub installs, it would help their higher-ups to be able to show their adoption on Steam Deck so far.

I also agree that it would be helpful to both app developers and Flathub to have a sense of the spread across distributions; e.g. if I as an app developer get a bug report from an Ubuntu LTS user that something isn't working right (e.g. due to a difference in Portals shipped in that version compared to what I'm testing on Fedora Silverblue), it makes a big difference if Ubuntu LTS users make up a large portion of my installs versus it being a one-off or tiny percent of my users affected.

**Comment by razzeee** [comment]
So I went hunting on where to handle this and it seems like it would need to be done here https://github.com/flatpak/flatpak/blob/main/common/flatpak-dir.c#L5393

**Comment by rany2** [comment]
If this issue is resolved it would help me in figuring out why some users are still using the old app id of the Ungoogled Chromium flatpak. I suspect that there are frontends that are ignoring the `eol-rebase` but I unfortunately cannot easily know the frontends that are at fault without having this metric.

At least right now, I only have the ostree/flatpak versions and these aren't as helpful as the distro name. I don't have a lot to work off to try and raise issues that would have these frontends respect eol-rebase.

**Comment by razzeee** [comment]
This would probably also be relevant due to the `usb` portal only working with flatpak `1.16.0` and greater. So that we can see, which distros we need to convince, to update.
